internal/app/api/types/message.go

```diff
@@ -15,3 +16,49 @@ type Message struct {
 	Attachments []string           `json:"attachments" bson:"attachments"`
 	CreateAt    time.Time          `json:"created_at" bson:"created_at"`
 }
+type PaginationMessage struct {
+	TotalItems      int `json:"totalItems"`
+	TotalPages      int `json:"totalPages"`
+	CurrentPage     int `json:"currentPage"`
+	MaxItemsPerPage int `json:"maxItemsPerPage"`
+}
+type ListMessageRes struct {
+	Content []*Message `json:"content"`
+}
+type GetListMessageRes struct {
+	PaginationMessage
+	ListMessageRes
+}
+
+type PagingNSortingMess struct {
+	Size int `json:"size" default:"100"`
+	Page int `json:"page" default:"1"`
+}
+
+func (ps *PagingNSortingMess) Init(page, size string) error {
+
+	if size == "" && page == "" {
+		ps.Size = 100
+		ps.Page = 1
+		return nil
+	}
+
+	sizeInt, err := strconv.ParseInt(size, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	if page == "" {
```

Comment:
What if size == "" and page == any number?

Reply:
like paginate user, the size not empty. the default value of page is 1. 

endpoint: /messages/612322427cc445b05c03264c => size=100, page=1
endpoint: /messages/612322427cc445b05c03264c?size=2 =>  size=2, page=1
endpoint: /messages/612322427cc445b05c03264c?page=2 =>  error , we need set size 

this was done in  https://github.com/paphuc/dating/pull/8